addons/src/Makefile.am

```diff
@@ -1,31 +1,4 @@
+include $(top_srcdir)/build/vars.init.mk
 include $(top_srcdir)/build/vars.build.mk
-
-geanyplugins_LTLIBRARIES = addons.la
-
-addons_la_SOURCES = \
-	addons.h \
-	ao_blanklines.h \
-	ao_doclist.h \
-	ao_openuri.h \
-	ao_systray.h \
-	ao_bookmarklist.h \
-	ao_markword.h \
-	ao_tasks.h \
-	ao_xmltagging.h \
-	ao_wrapwords.h \
-	ao_copyfilepath.h \
-	addons.c \
-	ao_blanklines.c \
-	ao_doclist.c \
-	ao_openuri.c \
-	ao_systray.c \
-	ao_bookmarklist.c \
-	ao_markword.c \
-	ao_tasks.c \
-	ao_xmltagging.c \
-	ao_wrapwords.c \
-	ao_copyfilepath.c
-
-addons_la_LIBADD = $(COMMONLIBS)
-
 include $(top_srcdir)/build/cppcheck.mk
```

Comment:
No, I mean that `make check` (from the top-level) should have the same behavior as running `make check` from each sub-directory.  As *build/cppcheck.mk* adds rules to be executed upon `make check`, and that this is only included in the recursive Makefile, these checks will only be run when running `make check` from the sub-directories and never from the top-level.

Before these changes, `make check` was recursive (like most common targets, like *all*), and so running it from the top-level ran checks in the whole tree.  Now, `cppcheck` checks are missing from all plugins you converted to non-recursiveness when invoked non-recursively.

You can probably understand quite easily by giving a look at *build/cppcheck.mk*, knowing that `check-local` is a dependency of the `check` target.